### PR TITLE
New Feature: HASH() adds hashing functions to dnsconfig.js language

### PR DESCRIPTION
**Pull request by fuero**
Having access to SHA1 would be great when managing catalog zones:

```javascript
foo_name_suffix = HASH("sha1", "foo.name") + "zones"
D("catalog.example!int"
    , REG_NONE
	, DnsProvider(my_pdns_int)
    , TXT("version", "2")
    , PTR(foo_name_suffix, "foo.name.")
    , A("primaries.ext." + foo_name_suffix, "192.168.1.1")
)
```

So this PR adds a `HASH` function to Javascript following the above example.

### CODE REVIEW

**Comment by tlimoncelli** [comment]
Sounds like a reasonable feature to add. 

Some changes I'd like to see in the PR:
* Move hashFunc to a separate file (hash.go)
* Add user-facing documentation? (https://docs.dnscontrol.org/developer-info/styleguide-doc#how-to-add-a-new-page)

**Comment by fuero** [comment]
Thanks for the feedback, I've added your suggestions.

**Comment by tlimoncelli** [comment]
Looks great!  Thanks for the fast response.

Please run `go generate ./...` to regenerate the docs. (I forgot to mention that)

CC @cafferata to review the docs.

**Review by tlimoncelli — APPROVED** [review]

**Comment by cafferata** [comment]
> CC @cafferata to review the docs.

@tlimoncelli, thanks for the mention.

@fuero, this new `hash()` function must be included within the table of content:
https://github.com/StackExchange/dnscontrol/blob/7b286091d5d15c771dc27428729e7f47db082622/documentation/SUMMARY.md#L22-L27

File: documentation/language-reference/top-level-functions/HASH.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+---
+name: HASH
+parameters:
+  - algorithm
+  - value
+parameter_types:
+  algorithm: string
```
Comment (cafferata) [inline documentation/language-reference/top-level-functions/HASH.md:7]: You could provide TypeScript autocomplete with the possible algorithm:

```suggestion
  algorithm: '"SHA1" | "SHA256" | "SHA512"'
```

**Comment by fuero** [comment]
@cafferata done, thanks!

**Review by cafferata — APPROVED** [review]

**Comment by tlimoncelli** [comment]
Looks good to me! Thanks for adding this feature!

**Pull request merged**